numpy/__init__.py

```diff
@@ -254,3 +254,33 @@ def _sanity_check():
 
     _sanity_check()
     del _sanity_check
+
+    def _mac_os_check():
+        """
+        Quick Sanity check for Mac OS look for accelerate build bugs. Testing numpy polyfit calls init_dgelsd(LAPACK)
+        """
+        try:
+            c = array([3., 2., 1.])
+            x = linspace(0, 2, 7)
+            y = polyval(c, x)
+            err = [1, -1, 1, -1, 1, -1, 1]
+            _ = polyfit(x, y + err, 2, cov=True)
+        except ValueError as err:
+            pass
```

Comment:
No need for the try/except, right? Accelerate generates a warning, not an error.

Reply:
There is a value error post the rank warning, hence catching the ValueError, we can also add the error message. The Value Error is a shape mismatch error, due to the failed dgelsd initialization.